File: JAVA/S1_BASIC/L5_BASICRECURSION/oneton.java
class recursion{
    void oneton(int n){
        if(n==0)return ;
        oneton(n-1);
        System.out.print(n+" ");
    }
}
public class oneton{
    public static void main(String[]args){
        recursion r=new recursion();
        r.oneton(5);
    }
}

File: python/S1_BASIC/L5_BASICRECURSION/oneton.py
class Recursion:
    def __init__(self,n):
        self.n=n
    def oneton(self,n):
        if n==0:
            return 
        # print(n,end=" ")
        self.oneton(n-1)
        print(n,end=" ")
if __name__=="__main__":
    n=int(input())
    R=Recursion(n)
    R.oneton(R.n)
        